proxy/proxy.go

```diff
@@ -395,7 +390,7 @@ func cloneHeaderExcluding(h http.Header, excludeList map[string]bool) http.Heade
 
 // copies a stream with flushing on every successful read operation
 // (similar to io.Copy but with flushing)
-func copyStream(to flusherWriter, from io.Reader, tracing *proxyTracing, span ot.Span) error {
+func copyStream(to flushedResponseWriter, from io.Reader, tracing *proxyTracing, span ot.Span) error {
```

Comment:
You could even keep the `flushWriter` here because its interface seems smaller than `flushedResponseWriter`. Makes  `copyStream` function more flexible. We just need Write and Flush?

Reply:
depends on the scope, but if we consider the entire proxy.go, then we need the full net/http.ResponseWriter, including the Header and WriteHeader functions